sopel/bot.py

```diff
@@ -360,34 +389,28 @@ def get_plugin_meta(self, name):
 
         return self._plugins[name].get_meta_description()
 
+    @deprecated(
+        reason="Replaced by specific `unregister_*` methods.",
+        version='7.1',
+        removed_in='7.2')
     def unregister(self, obj):
-        """Unregister a callable.
+        """Unregister a callable job or shutdown method.
 
         :param obj: the callable to unregister
         :type obj: :term:`object`
         """
-        if not callable(obj):
-            LOGGER.warning('Cannot unregister obj %r: not a callable', obj)
-            return
```

Comment:
We don't care anymore, because 1) the object must have specific attributes to be unregistered by this method and 2) this method is not used anymore by Sopel.